integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,408 @@
+            ruleList = new ArrayList<>(availableRules.values());
+        } else {
+            String[] parsedRuleStr = activeRuleStr.split(",");
+
+            for (String str : parsedRuleStr) {
+                addDataQualityRule(getAvailableRule(str.trim()));
+            }
+        }
+    }
+
+    /**
+     * Main method that conducts the data quality check. This is done by getting row counts and
+     * iterate through each data quality check. The results summary is provided after all are
+     * executed. The results are written to log as well as hive
+     * 
+     * @return Boolean value is the checks were successful
+     */
+    protected boolean doDataQualityChecks() {
+
+        boolean isSuccessful = false;
+        try {
+            String databaseName = flowAttributes.getAttributeValue(DataQualityConstants.CATEGORY_ATTRIBUTE);
+            String tableName = flowAttributes.getAttributeValue(DataQualityConstants.FEED_ATTRIBUTE);
+            String processingDttm = flowAttributes.getAttributeValue(DataQualityConstants.PROCESSING_DTTM_ATTRIBUTE);
+            
+            if (processingDttm.isEmpty()) {
+                String msg = "Required attribute processing_dttm is empty. Failing DQ";
+                log.error(msg);
+                throw new MissingAttributeException(msg);
+            }
+
+            String feedTableName = tableName + DataQualityConstants.FEED_TABLE_SUFFIX;
+            String invalidTableName = tableName + DataQualityConstants.INVALID_TABLE_SUFFIX;
+            String validTableName = tableName + DataQualityConstants.VALID_TABLE_SUFFIX;
+            String whereClause = DataQualityConstants.PROCESSING_DTTM_COLUMN + " = '" + processingDttm + "'";
+
+            setSourceRowCount();
+
+            long rowCount;
+            rowCount = getRowCount(databaseName, feedTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_FEED_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            rowCount = getRowCount(databaseName, invalidTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_INVALID_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            rowCount = getRowCount(databaseName, validTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_VALID_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            // Execute Data Quality rules
+            boolean rulePass = true;
+            boolean dqAllRulePass = true;
+            for (DataQualityRule rule : ruleList) {
+
+                log.info("Executing rule: " + rule.getName());
+
+                if (rule.loadAttributes(flowAttributes)) {
+                    rulePass = rule.evaluate();
+                } else {
+                    rulePass = false;
+                }
+
+                if (!rulePass) {
+                    log.error("Rule: " + rule.getName() + " has FAILED. Rule desc: " + rule.getDescription());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Rule: " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)